File: easy/add_binary_test.go
package easy

import (
	"testing"
)

func TestAddBinary(t *testing.T) {
	cases := []struct {
		a        string
		b        string
		expected string
	}{
		{"11", "1", "100"},
		{"1010", "1011", "10101"},
		{"0", "0", "0"},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := AddBinary(c.a, c.b)
			if result != c.expected {
				t.Errorf("AddBinary(%v, %v) == %v, expected %v", c.a, c.b, result, c.expected)
			}

		})

	}
}

File: src/test/java/easy/AddBinaryTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class AddBinaryTest {
  AddBinary instance = new AddBinary();

  @ParameterizedTest
  @MethodSource("dataAddBinary")
  public void testAddBinary(String a, String b, String expected) {

    String result = instance.solve(a, b);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataAddBinary() {
    return Stream.of(Arguments.of("11", "1", "100"), Arguments.of("1010", "1011", "10101"));
  }
}
